localstack-core/localstack/services/lambda_/event_source_mapping/pollers/kinesis_poller.py

```diff
@@ -200,3 +197,10 @@ def parse_data(self, raw_data: str) -> dict | str:
 
     def encode_data(self, parsed_data: dict) -> str:
         return base64.b64encode(json.dumps(parsed_data).encode()).decode()
+
+
+def has_data_filter_criteria_parsed(parsed_filters: list[dict]) -> bool:
```

Comment:
How come we are introducing this new function?

Reply:
inlined from legacy code removed here https://github.com/localstack/localstack/pull/11733/files#diff-7e0d439c4f8162098a496d085987f824965da98bdcdd38e983354c03c2c6fa21L232